### DISCUSSION TITLE
flavor of regex?

### DISCUSSION THREAD

**Discussion by jayveesea** [git-for-windows/git#5107] [Q&A] [answered]
What flavor of regex does git-for-windows use?  In particular I'm interested in using `git log -L` from Git Bash.

**Comment by dscho** [comment, marked as answer]
Git for Windows uses the same regular expressions as Git, which by default is the POSIX basic regular expression flavor. The [`-L :<funcname>:<path>`](https://git-scm.com/docs/git-log#Documentation/git-log.txt--Lltfuncnamegtltfilegt) option is parsed via [the `parse_range_funcname()` function](https://github.com/git/git/blob/master/line-range.c#L166-L246) which uses the standard POSIX [`regcomp()`/`regexec()`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/regcomp.html) functions.

Git for Windows relies on the gnulib implementation that is vendored into [`compat/regex/`](https://github.com/git/git/tree/v2.46.0/compat/regex).

Please note that [_Basic_ Regular Expressions](https://en.wikipedia.org/wiki/Regular_expression#POSIX_basic_and_extended) ("BRE") are used here, not POSIX Extended ones.

Also please note that [`--perl-regexp` has currently no effect on the `-L` option](https://github.com/gitgitgadget/git/issues/235).

**Reply by jayveesea** [reply]
thx!

what about `--perl-regexp` with `git grep`?

**Reply by dscho** [reply]
Yep, Git for Windows is compiled with PCRE2, therefore [the `--perl-regexp` option](https://git-scm.com/docs/git-grep#Documentation/git-grep.txt--P) will work for `git grep`.
